### PR TITLE
Add flag support for JSON deserializer

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR adds the flag support for the JSON deserializer and targets #361.
Please review carefully and ask me to add more tests if you like!

### CODE REVIEW

File: src/serialize/json/deserializer_callbacks.c [hunk 1]
```diff
@@ -266,6 +283,28 @@ yajl_string_cb(
...
+            }
+
+            char* buff = strndup((char*) str, len);
+            if (!buff) {
+                //!< @todo error, what now?
+                return 0;
+            }
+
+            ws_string_set_from_raw(state->register_name, buff);
+            free(buff); // `ws_string_set_from_raw()` does copying.
```
Comment (neithernut) [inline src/serialize/json/deserializer_callbacks.c]: Just allocate `buf` on the stack.

**Comment by neithernut** [comment]
`WS_TRANSACTION_FLAG_EXEC` sould be **set** by default.

**Comment by neithernut** [comment]
ACK for squashing.

File: src/serialize/json/deserializer_callbacks.c [hunk 2]
```diff
@@ -521,3 +568,27 @@ setup_transaction(
     return 0;
 }
 
+void
+finalize_message(
+    struct ws_deserializer* d
+) {
+    if (!d->buffer) {
+        // Do runtime check whether buffer object exists here, because _someone_
+        // desided against runtime checks in the utility functions
```
Comment (xVin) [inline src/serialize/json/deserializer_callbacks.c]: decided

**Comment by matthiasbeyer** [comment]
Fixed what @xVin annotated. I don't think there should be any further squashing.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**